Do not inject __init__.py into fixtures directories

The unit tests from `test/units/modules/system/interfaces_file/test_interfaces_file.py` seem to use files from `test/units/modules/system/interfaces_file/fixtures/input/` and  `test/units/modules/system/interfaces_file/fixtures/golden_output/`.

We inject `__init__.py` into `test/units/modules/system/interfaces_file/fixtures/input` (`fixtures` and all subdirs) and then the tests think it's a valid input file for a test.

As a result we get a permission denied error on `ansible_collections/ansible/misc/tests/unit/modules/system/interfaces_file/fixtures/golden_output/__init__.py_revert.exceptions.txt`.


Complete example (one of them):
```
___________________ TestInterfacesFileModule.test_no_changes ___________________
[gw2] linux -- Python 3.7.4 /tmp/python-rndnyy_e-ansible/python
self = <ansible_collections.ansible.misc.tests.unit.modules.system.interfaces_file.test_interfaces_file.TestInterfacesFileModule testMethod=test_no_changes>

    def test_no_changes(self):
        for testfile in self.getTestFiles():
            path = os.path.join(fixture_path, testfile)
            lines, ifaces = interfaces_file.read_interfaces_file(module, path)
>           self.compareInterfacesLinesToFile(lines, testfile)

tests/unit/modules/system/interfaces_file/test_interfaces_file.py:102: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/unit/modules/system/interfaces_file/test_interfaces_file.py:77: in compareInterfacesLinesToFile
    self.compareStringWithFile("".join([d['line'] for d in interfaces_lines if 'line' in d]), testname)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <ansible_collections.ansible.misc.tests.unit.modules.system.interfaces_file.test_interfaces_file.TestInterfacesFileModule testMethod=test_no_changes>
string = '', path = '__init__.py.test_no_changes'

    def compareStringWithFile(self, string, path):
        # self.assertEqual("","_",msg=path)
        testfilepath = os.path.join(golden_output_path, path)
        goldenstring = string
        if not os.path.isfile(testfilepath):
>           f = io.open(testfilepath, 'wb')
E           PermissionError: [Errno 13] Permission denied: '/root/ansible/ansible_collections/ansible/misc/tests/unit/modules/system/interfaces_file/fixtures/golden_output/__init__.py.test_no_changes'

tests/unit/modules/system/interfaces_file/test_interfaces_file.py:89: PermissionError
```

Needs further investigation.

Update: The command used for unit test run was `ansible-test units --requirements --python "3.7" -vvv --docker`

--- a/migrate.py
+++ b/migrate.py
@@ -705,13 +705,12 @@ def read_module_txt_n_fst(path):
 
 
 def inject_init_into_tree(target_dir):
-    for initpath in (
-            os.path.join(dp, '__init__.py')
-            for dp, dn, fn in os.walk(target_dir)
-            if '__init__.py' not in fn
-            # and any(f.endwith('.py') for f in fn)
-    ):
-        write_text_into_file(initpath, '')
+    """Insert ``__init__.py`` file into unit test directories."""
+    for root, dirs, files in os.walk(target_dir):
+        python_files_exist = any(fn.endswith('.py') for fn in files)
+        possibly_subpackages_exist = len(dirs) > 0
+        if '__init__.py' not in files and (python_files_exist or possibly_subpackages_exist):
+            write_text_into_file(os.path.join(root, '__init__.py'), '')
 
 
 def inject_readme_into_collection(collection_dir, *, ctx):

